aea/helpers/async_utils.py

```diff
@@ -0,0 +1,324 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the misc utils for async code."""
+import asyncio
+import datetime
+import logging
+import time
+from asyncio import CancelledError
+from asyncio.events import AbstractEventLoop, TimerHandle
+from asyncio.futures import Future
+from asyncio.tasks import ALL_COMPLETED, FIRST_COMPLETED, Task
+from threading import Thread
+from typing import (
+    Any,
+    Callable,
+    Coroutine,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+try:
+    from asyncio import create_task
+except ImportError:  # pragma: no cover
+    # for python3.6!
+    from asyncio import ensure_future as create_task  # type: ignore # noqa: F401
```

Comment:
I think it might be better to explicitly check for the python version using `sys`?

Reply:
I agree with David, the behaviour is more predictable (e.g. the import could fail for weird reasons). However, I'm not against this approach, I've seen it around quite often.